storagegroup/storagegroup.go

```diff
@@ -303,27 +261,5 @@ func WriteToObject(sg StorageGroup, o *objectSDK.Object) {
 
 	o.SetPayload(sgRaw)
 	o.SetType(objectSDK.TypeStorageGroup)
-
-	attrs := o.Attributes()
-	var expAttrFound bool
-
-	for i := range attrs {
-		if attrs[i].Key() == objectV2.SysAttributeExpEpoch {
-			expAttrFound = true
-			attrs[i].SetValue(strconv.FormatUint(sg.ExpirationEpoch(), 10))
-
-			break
-		}
-	}
-
-	if !expAttrFound {
-		var attr objectSDK.Attribute
-
-		attr.SetKey(objectV2.SysAttributeExpEpoch)
-		attr.SetValue(strconv.FormatUint(sg.ExpirationEpoch(), 10))
```

Comment:
Huh. This will break neofs-cli. I've not thought it's used to set _both_ specific and common one. Either it's OK (it can be set into `o` _before_ passing it into `WriteToObject`, so fixing neofs-cli shouldn't be hard) or we need `ExpirationEpoch` to work with the common attribute. Thoughts, opinions?

Reply:
but should SN accept SGs without expiration inside their bodies?

Reply:
Why not?

Reply:
> Why not?

cause that is how it should be in v2 API version (and there is no v3 yet) and SN has such validation now

Reply:
Then these methods can't be removed, right?

Reply:
@smallhive, let's keep ExpirationEpoch and SetExpirationEpoch with deprecated things inside. Sufficient for now, can be changed to work with outer expiration later if needed.

Reply:
> Then these methods can't be removed, right?

well, i am not sure whether it is a breaking change, we can discuss it: it had to be in the payload field and nobody knew that it could be OK to put expiration to the object's header field; then it was allowed to use header but all the repos in our stack _require_ an expiration to be set in both fields (btw, not sure we _had not had any SG objects_ by the time of this change in the public networks). so is it ok to allow an expiration to be set in the header field if our SNs can look for it in both fields? what if they differ? removing completely looks nice but not sure we can do that

cc @cthulhu-rider

Reply:
Dropped the commit